packages/gatsby-source-contentful/src/extend-node-type.js

```diff
@@ -627,6 +678,74 @@ exports.extendNodeType = ({ type, store, cache, getNodesByType }) => {
     })
   }
 
+  const getDominantColor = async ({ image, options }) => {
+    try {
+      const { rgbToHex } = require(`gatsby-plugin-sharp`)
+      const sharp = require(`gatsby-plugin-sharp/safe-sharp`)
+
+      const absolutePath = await cacheImage(store, image, options)
+
+      const pipeline = sharp(absolutePath)
+      const { dominant } = await pipeline.stats()
+
+      // Fallback in case sharp doesn't support dominant
+      const dominantColor = dominant
+        ? rgbToHex(dominant.r, dominant.g, dominant.b)
+        : `rgba(0,0,0,0.5)`
+
+      dominantColorCache.set(absolutePath, dominantColor)
+
+      return dominantColor
+    } catch (e) {
+      console.error(
+        `[gatsby-source-contentful] Please install gatsby-plugin-sharp`
+      )
+      return `rgba(0,0,0,0.5)`
+    }
```

Comment:
Any reason we're not exposing this from sharp?

Reply:
That's a good point. The answer is "because nothing else has needed it before", but yes a standard approach is a good idea. I'll make that change